### PR TITLE
Convert vnetpeerings service to SDKv2

### LINKED ISSUE #3926: Update vnetpeerings service to SDK v2

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Converts the vnetpeerings service to azure-sdk-for-go version 2 and the `ascynpoller` framework for long-running operations.

**Which issue(s) this PR fixes**:

Fixes #3926

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by nojnhuh — COMMENTED** [review]
/lgtm

File: azure/services/vnetpeerings/client.go [hunk 1]
```diff
@@ -18,139 +18,103 @@ package vnetpeerings
...
 
 	ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultAzureCallTimeout)
 	defer cancel()
 
-	err = deleteFuture.WaitForCompletionRef(ctx, ac.peerings.Client)
+	pollOpts := &runtime.PollUntilDoneOptions{Frequency: asyncpoller.DefaultPollerFrequency}
+	_, err = poller.PollUntilDone(ctx, pollOpts)
 	if err != nil {
-		// if an error occurs, return the future.
+		// if an error occurs, return the Poller.
```
Comment (willie-yao) [inline azure/services/vnetpeerings/client.go:113]: nit
```suggestion
		// if an error occurs, return the poller.
```

Comment (mboersma) [reply azure/services/vnetpeerings/client.go]: Done!

**Pull request merged**